src/features/MoviesBrowser/PopularMovies/index.js

```diff
@@ -9,19 +9,32 @@ import MainHeader from "../../../common/MainHeader";
 
 const PopularMovies = () => {
   const [currentPage, setCurrentPage] = useState(1);
+  const [isLoading, setIsLoading] = useState(true);
   const moviesData = useMoviesData(currentPage);
 
   const totalPages = () =>
     moviesData.total_pages > 500 ? 500 : moviesData.total_pages;
+
   const whichPage = (page) => {
     if (page >= 1 && page <= totalPages()) {
+      setIsLoading(true);
       setCurrentPage(page);
     }
   };
 
+  useEffect(() => {
+    const loadingTimeout = setTimeout(() => {
+      setIsLoading(false);
+    }, 1000);
+
+    return () => {
+      clearTimeout(loadingTimeout);
+    };
+  }, [currentPage]);
+
   return (
     <>
-      {moviesData.status === "loading" ? (
+      {isLoading ? (
         <Loading />
       ) : moviesData.status === "error" ? (
         <Error />
```

Comment:
Wouldn't it be better to do it inside of the useMovieData custom hook just like you've done it with Error?

Reply:
Agreed !